lib/core/src/Cardano/Wallet.hs

```diff
@@ -904,61 +920,70 @@ readNextWithdrawal
         )
     => ctx
     -> WalletId
+    -> Quantity "lovelace" Word64
     -> IO (Quantity "lovelace" Word64)
-readNextWithdrawal ctx wid = db & \DBLayer{..} -> do
-    (pp, withdrawal) <- atomically $ (,)
-        <$> readProtocolParameters pk
-        <*> fmap getQuantity (readDelegationRewardBalance pk)
-    case pp of
+readNextWithdrawal ctx wid (Quantity withdrawal) = db & \DBLayer{..} -> do
+    liftIO (atomically $ readProtocolParameters $ PrimaryKey wid) <&> \case
         -- May happen if done very early, in which case, rewards are probably
         -- not woth considering anyway.
-        Nothing -> pure (Quantity 0)
-
-        Just ProtocolParameters{txParameters} -> do
+        Nothing -> Quantity 0
+        Just ProtocolParameters{txParameters} ->
             let policy = W.getFeePolicy txParameters
 
-            let costOfWithdrawal =
+                costOfWithdrawal =
                     minFee policy (mempty { withdrawal })
                     -
                     minFee policy mempty
 
-            pure $ if toInteger withdrawal < 2 * costOfWithdrawal
+            in
+                if toInteger withdrawal < 2 * costOfWithdrawal
```

Comment:
could you recall me why we have here `2 * costOfWithdrawal` , ie., why `2` ?

Reply:
The rationale was that: "withdrawal should bring **at least much value** as they cost.". 

This is to prevent withdrawals from either loosing money, or being consumed too early when they're really small (imagine an account with `0.31` Ada, and the withdrawal cost `0.3` Ada. If we were just ensuring the withdrawal to be `> costOfWithdrawal` (without `2 *`), then the resulting withdrawal would be `0.01` Ada :/ .. And it really feels not worth it given the fees. Hence the times two.